spec/features/visitors/sign_up_spec.rb

```diff
@@ -301,4 +305,38 @@
     expect(page).to have_content 'Confirmation token is invalid'
     expect(current_path).to eq user_confirmation_path
   end
+
+  # Scenario: Visitor can be informed that they've already confirmed
+  #   Given I've confirmed my email and created a password, but forgot that I did
+  #   When I ask for confirmation instructions
+  #   Then I receive an email letting me know I've already confirmed
+  scenario 'visitor is notified via email when they have already confirmed', email: true do
+    sign_up_with_and_set_password_for('test@example.com')
+
+    visit destroy_user_session_url
+
+    click_link "Didn't receive confirmation instructions?"
+    fill_in 'Email', with: 'test@example.com'
+    click_button 'Resend confirmation instructions'
+
+    expect(last_email.body).to match(/already been confirmed/)
+    expect(last_email.subject).to eq t('upaya.mailer.already_confirmed.subject')
+  end
+
+  # Scenario: Confirmed visitor confirms again while signed out
+  #   Given I've confirmed my email, created a password, and signed out
+  #   When I click the confirmation link in the email again
+  #   Then I see a message that I've already confirmed
+  #   And I am redirected to the sign in page
+  scenario 'confirmed visitor clicks confirmation link again while signed out' do
```

Comment:
Followup on earlier comment: Seems that some scenarios describe purely what the user does while others describe the expected action.. do we prefer one of these styles over the other?